src/server/req_signal.c

```diff
@@ -112,19 +113,20 @@ req_signaljob(struct batch_request *preq)
 	if (parent == (job *)0)
 		return;		/* note, req_reject already called */
 
-	if ((strcmp(preq->rq_ind.rq_signal.rq_signame, SIG_RESUME) == 0) ||
-		(strcmp(preq->rq_ind.rq_signal.rq_signame, SIG_SUSPEND) == 0)) {
+	if (strcmp(preq->rq_ind.rq_signal.rq_signame, SIG_RESUME) == 0 || strcmp(preq->rq_ind.rq_signal.rq_signame, SIG_ADMIN_RESUME) == 0)
+		susrsm = -1;
+	else if (strcmp(preq->rq_ind.rq_signal.rq_signame, SIG_SUSPEND) == 0 || strcmp(preq->rq_ind.rq_signal.rq_signame, SIG_ADMIN_SUSPEND) == 0)
+		susrsm = 1;
+
+	
+	if (susrsm) {
```

Comment:
Why add a check for susrsm and not do it for any signal? Granted that probably only the above signals can come down so far, but even in that case the check is redundant. In fact, the always reject is better lest it falls through to after the reject?